### DISCUSSION TITLE
Dent-Clack Backend

### DISCUSSION THREAD

**Discussion by Grey-A** [pythoniclaver/Dent-Clack#50] [Polls]
So regarding the backend i can do it two ways, i can just make it a normal django project where ill connect all the templates to their respective url paths but in doing so ill change the entire file structure.
Or i can make an API where you guys can connect with it from the frontend.

NOTE: developers need to install Django either way because it is required if you want to make any backend interactions 

##### Benefits of using a normal Django project approach
- Contributors don't have to do anything too technical they will just keep doing what they do just that the file structure will change
- Since we aren't using a js framework, using an API might be overkill and it will add another layer of complexity for others

##### Disadvantages of using the Django project approach
- Although it wont change any frontend things, using this approach might warrant extra interference from the backend developer for matters regarding any interactions with the backend
- The file structure will change and it might warrant we alter the pointers to the js and css files in our html file (script tags etc)

##### Benefits of using an API
- By using an API we don't have to alter the file structure 
- It makes a very minimal amount of change to the entire project's form 

##### Disadvantages of using an API
- More JavaScript has to be written in order to connect to the API and interpret the information 
- Not very beginner friendly for those who aren't very familiar with javascript json interactions 

PERSONAL THOUGHTS:  I think the API will be better but on the premise that the contributors are familiar with Javascript API interactions 

A simpler way to view this is:
1. Normal project method will increase the backend's work
2. API method will increase the frontend's work

**Comment by Grey-A** [comment]
dont forget to vote ladies and gentlemen

**Comment by vin-jex** [comment]
I love this man but I need to clarify some misunderstanding, the version1 of this project was built just to help people participate in hacktoberfest but the Version2 >>> Main version would be built in the recommended and best way.

We're going to use Reactjs, tailwind, typescript and any other library that would make our life easier for the frontend.

For our contributors that stand out, we're going to give them the opportunity to also work with us on the main version, that's the purpose of this open source.

**Comment by vickyaryan7** [comment]
I think API is the best stuff to do
